### ISSUE TITLE
[Bug]: Waypoint not persistant

### ISSUE THREAD

**Issue by mattba10** [TeamJM/journeymap#773]
### What happened?

A bug happened!
Playing an instanced server of Monifactory version 0.7.7. every server restart waypoints are wiped. I don't know much more except this following anecdote from another user;
"I figured out why this is happening. Instance data is not saved in JM. JM thinks you're in a new world when the server restarts, therefore will start making new data.
I find this behaviour very annoying, but I'm not sure it can be fixed. This can be very taxing on your drive's storage, so make sure to clear this directory every now and then"
![image](https://github.com/user-attachments/assets/9fd02b83-af32-4ede-86e8-30f7632bd37d)
![image](https://github.com/user-attachments/assets/8ce3c06a-5ae2-4882-8294-974053b90675)

### Mod Loader

Forge

### Minecraft Version

1.20.1

### Version

5.9.x

### Relevant log output

[journeymap.log](https://github.com/user-attachments/files/16627376/journeymap.log)
[latest.log](https://github.com/user-attachments/files/16627399/latest.log)
(Don't think either of these will help but I did it anyway)

**Comment by mysticdrew** [comment]
Have the server admins disable worldId, it is likely an issue with your server's host not keep the dat file that stores the ID so a new one is generated when the server restarts.

**Comment by TheFlash787** [comment]
@mysticdrew Admin here, where would you disable world ID? 

We are running multiple Monifactory servers and players would like to keep their waypoints synced between them locally. Ideally, storing based off the IP only, such as `monifactory.modrealms.net` would be great.

**Comment by mysticdrew** [comment]
World id is in the Server Admin screen. 
https://teamjm.github.io/journeymap-docs/latest/Server%20Docs/settings/global-properties/
WorldId is used to identify multiplayer worlds uniquely, specifically for worlds that use multiple world setups. The file that stores the worldId is in `minecraft/world/data/WorldUUID.dat` if using forge/fabric/neoforge. 

If you are using the plugin `minimap-control` on a bukkit/spigot/paper server, they have a bug in their latest version that sends a new world ID for every dimension, not just world. 

If users want to save by IP, that option is only available in JourneyMap 6.0.0 for MC 1.21+ and it has to be enabled manually in the journeymap settings `Client > Advanced Options`

**Comment by VibrantRaven** [comment]
Hey @mysticdrew 

Do you know where the JourneyMap data typically is stored, or if it uses the global level.dat in some way or another?

**Comment by mysticdrew** [comment]
> Hey @mysticdrew
> 
> Do you know where the JourneyMap data typically is stored, or if it uses the global level.dat in some way or another?

why are you piggy backing this issue for a question like this? If you look at the original issue here they have the path where journeymap data is stored.

**Comment by VibrantRaven** [comment]
> > Hey @mysticdrew
> > Do you know where the JourneyMap data typically is stored, or if it uses the global level.dat in some way or another?
> 
> why are you piggy backing this issue for a question like this? If you look at the original issue here they have the path where journeymap data is stored.

As to why I am "piggy backing" on this specific issue, is because I'm a developer alongside the Admin above so it's completely relevant.

I know where data specifically is stored, I was however hoping to see that it had some form of connection to the level.dat file as that'd be an easy fix as your solution for 1.21+ isn't sufficient.

**Comment by mysticdrew** [comment]
![image](https://github.com/user-attachments/assets/744061c6-3c7b-4343-9b03-7c0e4e4d19df)

**Comment by VibrantRaven** [comment]
> You didn't state that, just sounded like a random question on a random issue. (this happens a lot) what data are you talking about then? We don't use level.dat for anything I don't know what versions of anything you guys are using.

My apologies, I should have been clear on that. We're currently using the one shipped with the Modpack "Monifactory" which is JourneyMap 1.20.1 v5.9.21. If however it doesn't have any relevance to the level.dat, the WorldUUID.dat might suffice in some way or another to the matter.

**Comment by mysticdrew** [comment]
I have seen similar things like this happen in the past when the server did not have permissions to create files or the disk is full. 

if worldId is enabled
So when a player logs in for the first time after a restart. JourneyMap checks, do we have a worldUUID.dat? 
If yes, read it, send the ID to the player, store the uuid in memory (it never changes or gets updated after initial load). 
If no, create a random uuid, store it in memory, send to player, then save it to disk(This is where the problem is likely happening). 

One thing to look at is are there any exceptions in the logs on the server when the first player with journeymap logs in relating to saving the file.

**Comment by mysticdrew** [comment]
Minecraft does not have unique identifier for servers as seed is not viable, so we generate it for the server. 
Each server instance has one, and it is used for all dimensions on that server. 

So if you have multiple server instances, each instance will have a worldid that we generate. 

My guess is you are using some proxy system to transfer players and their sessions to each server?

**Comment by TheFlash787** [comment]
@mysticdrew I've found a solution that works for our case, we can simply persist our `overworld/data/WorldUUID.dat` file as this was previously not persistent between server restarts. While the player worlds were persisting their waypoints, the actual server's overworld was not -- thus giving the player a fresh WorldUUID each time.

This has now been resolved, thanks so much for the additional context & assistance!
